src/Pagination/PaginateDirective.php

```diff
@@ -112,19 +113,21 @@ public function resolveField(FieldValue $fieldValue): FieldValue
 
                 $query = $builderResolver($root, $args, $context, $resolveInfo);
                 assert(
-                    $query instanceof QueryBuilder || $query instanceof EloquentBuilder || $query instanceof Relation,
+                    $query instanceof QueryBuilder || $query instanceof EloquentBuilder || $query instanceof ScoutBuilder || $query instanceof Relation,
                     "The method referenced by the builder argument of the @{$this->name()} directive on {$this->nodeName()} must return a Builder or Relation."
                 );
             } else {
                 $query = $this->getModelClass()::query();
             }
 
-            $query = $resolveInfo
-                ->argumentSet
-                ->enhanceBuilder(
-                    $query,
-                    $this->directiveArgValue('scopes') ?? []
-                );
+            if (! $query instanceof ScoutBuilder) {
```

Comment:
This would skip possibly important steps in https://github.com/pyrou/lighthouse/blob/6a810c0b00aa917b6ceb45c52d4080ba4268a5a9/src/Scout/ScoutEnhancer.php#L99-L106

Reply:
Would be ok if I made change so Nuwave\Lighthouse\Scout\ScoutEnhancer accept also ScoutBuilder instances (see https://github.com/pyrou/lighthouse/commit/233dd7827c3328056e222bc0598d5c2d591e6978) ?

(PaginateDirective and AllDirective still has to be modified of course)

Reply:
Yeah, this is basically what I suggested in https://github.com/nuwave/lighthouse/pull/2139#issuecomment-1123738101.